radicle-httpd/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use std::{net, process};
+
+use tracing::dispatcher::Dispatch;
+
+use radicle_httpd as httpd;
+
+#[derive(Debug)]
+pub struct Options {
+    pub listen: net::SocketAddr,
+}
+
+impl Options {
+    fn from_env() -> Result<Self, lexopt::Error> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_env();
+        let mut listen = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("listen") => {
+                    let addr = parser.value()?.parse()?;
+                    listen = Some(addr);
+                }
+                Long("help") => {
+                    println!("usage: radicle-httpd [--listen <addr>]");
+                    process::exit(0);
+                }
+                _ => return Err(arg.unexpected()),
+            }
+        }
+        Ok(Self {
+            listen: listen.unwrap_or_else(|| ([0, 0, 0, 0], 8080).into()),
+        })
+    }
+}
+
+impl From<Options> for httpd::Options {
+    fn from(other: Options) -> Self {
+        Self {
+            listen: other.listen,
+        }
+    }
+}
+
+#[cfg(feature = "logfmt")]
+mod logger {
+    use tracing_subscriber::layer::SubscriberExt as _;
+    use tracing_subscriber::EnvFilter;
+
+    pub fn subscriber() -> impl tracing::Subscriber {
+        tracing_subscriber::Registry::default()
+            .with(EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")))
+            .with(tracing_logfmt::layer())
+    }
+}
+
+#[cfg(not(feature = "logfmt"))]
+mod logger {
+    pub fn subscriber() -> impl tracing::Subscriber {
+        tracing_subscriber::FmtSubscriber::new()
+    }
+}
```

Comment:
Curious, what's the purpose for the logfmt feature?

I'm also a little concerned about the long term maintenance burden to support different features.

Reply:
It's what we use when we want the logs to be parsed for grafana dashboards for example.